openedge-plugin/src/main/java/org/sonar/plugins/openedge/foundation/OpenEdgeComponents.java

```diff
@@ -122,38 +115,37 @@ private void registerLicences(LicenceRegistrar[] licRegistrars, String permanent
       LOG.debug("Found {} licence - Permanent ID '{}' - Customer '{}' - Repository '{}' - Expiration date {}",
           lic.getType().toString(), lic.getPermanentId(), lic.getCustomerName(), lic.getRepositoryName(),
           DateFormat.getDateTimeInstance(DateFormat.LONG, DateFormat.LONG).format(new Date(lic.getExpirationDate())));
-      if (!lic.getPermanentId().isEmpty() && !permanentId.equals(lic.getPermanentId())) {
-        LOG.debug("Skipped licence as it doesn't match permanent ID '{}'", permanentId);
-        continue;
-      }
-      // Licence with highest expiration date wins
-      Licence existingLic = licences.get(lic.getRepositoryName());
-      if ((existingLic == null) || (existingLic.getExpirationDate() < lic.getExpirationDate())) {
-        licences.put(lic.getRepositoryName(), lic);
-        LOG.debug("Installed !");
-      } else {
-        LOG.debug("Conflict, skipped licence");
+
+      // Only one licence per repository / permID
+      Licence existingLic = getLicence(lic.getRepositoryName(), lic.getPermanentId());
+      if (existingLic == null) {
+        licences.add(lic);
+      } else if (existingLic.getExpirationDate() < lic.getExpirationDate()) {
+        licences.remove(existingLic);
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') This call to "remove()" may be a performance hot spot if the collection is large. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=squid%3AS2250)